projects/packages/my-jetpack/src/products/class-stats.php

```diff
@@ -145,21 +160,46 @@ public static function get_wpcom_free_product_slug() {
 	 * @return boolean
 	 */
 	public static function has_required_plan() {
-		$purchases_data = Wpcom_Products::get_site_current_purchases();
-		if ( is_wp_error( $purchases_data ) ) {
-			return false;
-		}
-		if ( is_array( $purchases_data ) && ! empty( $purchases_data ) ) {
-			foreach ( $purchases_data as $purchase ) {
-				if ( str_starts_with( $purchase->product_slug, 'jetpack_stats' ) ) {
-					return true;
-				}
-				if ( str_starts_with( $purchase->product_slug, 'jetpack_complete' ) ) {
-					return true;
+		return true;
+	}
+
+	/**
+	 * Gets the 'status' of the Stats product
+	 *
+	 * @return string
+	 */
+	public static function get_status() {
```

Comment:
> FWIW, I am able to access the detailed stats page with just a site connection while testing?

Hmm, on my local Jetpack dev env as well as a Jurassic Ninja site, starting off as connected, if I click the "manage connection" link from either the Jetpack Dashboard or from the bottom of My Jetpack, then click "Disconnect". Now (while not connected) if I try to load page: `/wp-admin/admin.php?page=stats`, I get a page that says, "Sorry, you are not allowed to access this page." (See screenshot) ⤵️

![Markup 2024-02-05 at 17 28 23](https://github.com/Automattic/jetpack/assets/11078128/39ca2a40-757b-453c-a2a9-971685c3d45e)

JN site
![Markup 2024-02-05 at 18 18 40](https://github.com/Automattic/jetpack/assets/11078128/8b3ef459-f713-4ff7-a4ff-96ed348cc3bb)

Maybe I'm misunderstanding the meaning of "site connection"?  What Steps are you taking to establish only a "site connection"?

Reply:
> Hmmm, It feels a bit strange to duplicate the get_status function here in the child class 🤔
>
> Is this replicated here to handle the user connection condition differently?
>
> What happens if we remove this and set $requires_user_connection back to false? If the user goes to "view detailed stats" and are not connected, could they complete the connection from the detailed stats page?

Yes, with the goal of making the "Learn more" button show when there is no connection (just like the other product cards), the way the `get_status()` function works is, when `$requires_user_connection` is `false` then `get_status()` will check `has_required_plan()` and if it's `false`, the "Learn more" button shows.  This works for all the other cards because `has_required_plan()` returns `false` for the other products, I believe, because they all require a plan to work.  Stats is different because the only thing required for Stats to work (currently) is a connection (not a plan).  That being said, for Stats to show the "Learn more" button when there's no connection, we either have to, 1: Set `$requires_user_connection` to `false` and then return false in `has_required_plan()` when there is no connection (what I did previously, in the first revision). OR 2: We set `$requires_user_connection` to `true` but then have `get_status()` return '`needs_purchase_or_free`' instead of '`error`' when there is no connection (what I did in this revision). Which incidentally, I found a cleaner way to do this, rather than replicating (with a minor change) the whole `get_status()` function, which is:
```
public static function get_status() {
	$status = parent::get_status();
	if ( $status === 'error' ) {
		$status = 'needs_purchase_or_free';
	}
	return $status;
}
```

Reply:
> Maybe I'm misunderstanding the meaning of "site connection"? What Steps are you taking to establish only a "site connection"?

Good question. So, to get a site-only connection is a bit of a workaround at the moment.

Start on the connection screen like normal and click the button to connect.
![Screenshot 2024-02-06 at 8 37 13 AM](https://github.com/Automattic/jetpack/assets/18016357/a56f34f4-07c4-4c0d-8d3f-d3c30cc59310)

Wait until you see the user connect screen, but do not authorize the user connection. Navigate back to the site instead. Now, at the bottom of My Jetpack, you should see that the site is connected, but the user is not.
![Screenshot 2024-02-06 at 8 39 51 AM](https://github.com/Automattic/jetpack/assets/18016357/713dc910-e08e-47f8-a872-6a2dd55bc926)

The site is now "site connected" and has access to some Jetpack features, which should include the stats page.